theano/sandbox/gpuarray/multinomial.py

```diff
@@ -0,0 +1,256 @@
+# TODO test dtype != float32
+import os
+
+import pygpu
+
+import theano
+import theano.sandbox.multinomial
+from theano import Apply, config
+from theano.gof import Op, local_optimizer
+from theano.tensor import NotScalarConstantError, get_scalar_constant_value
+from theano.sandbox import gpuarray
+from .basic_ops import as_gpuarray_variable, infer_context_name
+from .fp16_help import write_w
```

Comment:
Part of the multi-dtype, float16 support it done, but not all. I let the finishing of this to another PR, so we can deprecate the old back-end earlier.